cmd/README.md

```diff
@@ -0,0 +1,639 @@
+```
+
+We would have modify the `Equal()` function as follows: 
+
+![](2022-09-01-22-54-22.png)
+
+The next place we see the `CUSTOM CODE STARTS HERE` is in `Configure()`
+Let's set it up. 
+
+```
+// Configure configures [state] with the initial configuration.
+func (c *HelloWorldConfig) Configure(_ ChainConfig, state StateDB, _ BlockContext) {
+	// CUSTOM CODE STARTS HERE
+	// This will be called in the first block where HelloWorld stateful precompile is enabled.
+	// 1) If BlockTimestamp is nil, this will not be called
+	// 2) If BlockTimestamp is 0, this will be called while setting up the genesis block
+	// 3) If BlockTimestamp is 1000, this will be called while processing the first block
+	// whose timestamp is >= 1000
+	//
+	// Set the initial value under [common.BytesToHash([]byte("storageKey")] to "Hello World!"
+	res := common.LeftPadBytes([]byte("Hello World!"), common.HashLength)
+	state.SetState(IHelloWorldAddress, common.BytesToHash([]byte("storageKey")), common.BytesToHash(res))
+}
+```
+
+Next place to modify is in our `sayHello()` function.
+
+```
+func sayHello(accessibleState PrecompileAccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
+	if remainingGas, err = deductGas(suppliedGas, SayHelloGasCost); err != nil {
+		return nil, 0, err
+	}
+	if readOnly {
+		return nil, remainingGas, vmerrs.ErrWriteProtection
+	}
+	// no input provided for this function
+
+	// CUSTOM CODE STARTS HERE
+	// Get the current state
+	currentState := accessibleState.GetStateDB()
+	// Get the value set at recipient
+	value := currentState.GetState(IHelloWorldAddress, common.BytesToHash([]byte("storageKey")))
+	// Do some processing and pack the output
+	packedOutput, err := PackSayHelloOutput(string(common.TrimLeftZeroes(value.Bytes())))
+	if err != nil {
+		return nil, remainingGas, err
+	}
+
+	// Return the packed output and the remaining gas
+	return packedOutput, remainingGas, nil
+}
+```
+
+Finally we can modify our `setGreeting()` function 
+
+```
+func setGreeting(accessibleState PrecompileAccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error) {
+	if remainingGas, err = deductGas(suppliedGas, SetGreetingGasCost); err != nil {
+		return nil, 0, err
+	}
+	if readOnly {
+		return nil, remainingGas, vmerrs.ErrWriteProtection
+	}
+	// attempts to unpack [input] into the arguments to the SetGreetingInput.
+	// Assumes that [input] does not include selector
+	// You can use unpacked [inputStruct] variable in your code
+	inputStr, err := UnpackSetGreetingInput(input)
+	if err != nil {
+		return nil, remainingGas, err
+	}
+
+	// CUSTOM CODE STARTS HERE
+	// setGreeting is the execution function of "SetGreeting(name string)" and sets the storageKey in the string returned by hello world
+	res := common.LeftPadBytes([]byte(inputStr), common.HashLength)
```

Comment:
one note: if the string is longer than 32 bytes, then it will end up getting cropped from the left here, so we can either leave this as is or add a check for the length of the string to make sure it returns an error instead if the string will be cropped.

Reply:
Ah ok